### DISCUSSION TITLE
Fix grain reconstruction at edge of map

### DISCUSSION THREAD

**Discussion by macosta541** [mtex-toolbox/mtex#1423] [Ask Anything] [answered]
Hi everyone,

I have a map that for some reason has issues with grain reconstructions at the edges.
Here is the code that I am currently using. 
Help is greatly appreciated:

`% grain reconstruction
grains = calcGrains(ebsd('indexed'), 'threshold', 10*degree); % reconstruct grains w/threshold angle of 10 degrees
grains = smooth(grains, 5)% smooth grains

% make grain plots
figure()
plot(grains) % individual grains of all indexed phases
hold on
plot(grains.boundary, 'color', 'black')`

![badGrainReconstruction](https://user-images.githubusercontent.com/74112001/166643594-f2a73428-2106-4de2-9e1b-f159dc138f1b.jpg)
![ebsd](https://user-images.githubusercontent.com/74112001/166644081-53985ff0-198f-405b-aa62-b71ff3a2176f.jpg)

**Comment by lfgmorales** [comment, marked as answer]
hi


you may want to add the options 'boundary', 'tight', in your grains calculation


something like


grains = calcGrains(ebsd('indexed'),'boundary','tight');

ypu may also want to have a look here


https://mtex-toolbox.github.io/GrainReconstruction.html



cheers


L


Luiz F. G. Morales
ScopeM & Structural Geology and Tectonics Group
ETH Zürich
HPM C57.3

Otto-Stern-Weg 3

8093 Zürich

Switzerland

Phone: + 41 44 633 3746

<tel:+41446333746>E-mail: ***@***.***



"No trees were destroyed in the sending of this message, however, a significant number of electrons were terribly inconvenienced"


________________________________
From: macosta541 ***@***.***>
Sent: 04 May 2022 10:14:04
To: mtex-toolbox/mtex
Cc: Subscribed
Subject: [mtex-toolbox/mtex] Fix grain reconstruction at edge of map (Discussion #1423)


Hi everyone,

I have a map that for some reason has issues with grain reconstructions at the edges.
Here is the code that I am currently using.
Help is greatly appreciated:

`% grain reconstruction
grains = calcGrains(ebsd('indexed'), 'threshold', 10*degree); % reconstruct grains w/threshold angle of 10 degrees
grains = smooth(grains, 5)% smooth grains

% make grain plots
figure()
plot(grains) % individual grains of all indexed phases
hold on
plot(grains.boundary, 'color', 'black')`

[badGrainReconstruction]<https://user-images.githubusercontent.com/74112001/166643594-f2a73428-2106-4de2-9e1b-f159dc138f1b.jpg>
[ebsd]<https://user-images.githubusercontent.com/74112001/166644081-53985ff0-198f-405b-aa62-b71ff3a2176f.jpg>

—
Reply to this email directly, view it on GitHub<https://github.com/mtex-toolbox/mtex/discussions/1423>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ALE5INZHO3Z7FQUOVB42FVDVIIWUZANCNFSM5VBF3K5Q>.
You are receiving this because you are subscribed to this thread.Message ID: ***@***.***>

[ { ***@***.***": "http://schema.org", ***@***.***": "EmailMessage", "potentialAction": { ***@***.***": "ViewAction", "target": "https://github.com/mtex-toolbox/mtex/discussions/1423", "url": "https://github.com/mtex-toolbox/mtex/discussions/1423", "name": "View Discussion" }, "description": "View this Discussion on GitHub", "publisher": { ***@***.***": "Organization", "name": "GitHub", "url": "https://github.com" } } ]

**Reply by macosta541** [reply]
Exactly what I was looking for, thank you Luiz!
